api/api/views/media_views.py

```diff
@@ -99,8 +119,63 @@ def retrieve(self, request, *_, **__):
         return Response(serializer.data)
 
     def list(self, request, *_, **__):
-        params = self._get_request_serializer(request)
+        params = self._get_request_serializer(request, strategy="search")
+        return self.get_media_results(request, "search", params)
+
+    def collection(self, request, tag, source, creator, *_, **__):
+        if tag:
+            collection_params = {"tag": [tag]}
+        elif creator:
```

Comment:
I think the `+` in the regex prevents an empty source in the URL:

```
>>> path = re.compile(r"source/(?P<source>[^/.]+)")
>>> path.match("source/") is None
True
>>> path.match("source/hello") is None
False
>>> path.match("source//") is None
True
```

Same goes for the other collection parameters. In the logic of this function, I think it's safe to just `else` without checking for `source`. These are less of _existence_ checks and more _inferences_ of the strategy based on the parameters.

That _might_ be clearer if this method also accepted a `collection_type` parameter of `creator | source | tag`, but I'm ambivalent. Alternatively, each action handler could construct the `collection_params` itself and pass it already initialised to `::collection`. That would leave collection completely (blissfully) ignorant of these details. If anywhere _should_ check that `tag`, etc aren't empty, it would be the action handler, I would think, not a sub-function of it.